cmd/root.go

```diff
@@ -0,0 +1,148 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/adrg/xdg"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"sigs.k8s.io/hydrophone/pkg/client"
+	"sigs.k8s.io/hydrophone/pkg/common"
+	"sigs.k8s.io/hydrophone/pkg/service"
+)
+
+var (
+	cfgFile          string
+	kubeconfig       string
+	parallel         int
+	verbosity        int
+	outputDir        string
+	cleanup          bool
+	listImages       bool
+	conformance      bool
+	focus            string
+	skip             string
+	conformanceImage string
+	busyboxImage     string
+	dryRun           bool
+	testRepoList     string
+	testRepo         string
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "hydrohpone",
+	Short: "Hydrophone is a lightweight runner for kubernetes tests.",
+	Long:  `Hydrophone is a lightweight runner for kubernetes tests.`,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if conformance && focus != "" {
+			conformance = false
+			focus = ""
+			log.Fatal("specify either --conformance or --focus arguments, not both")
+			os.Exit(1)
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		client := client.NewClient()
+		config, clientSet := service.Init(viper.GetString("kubeconfig"))
+		client.ClientSet = clientSet
+		common.PrintInfo(client.ClientSet, config)
+		if cleanup {
+			service.Cleanup(client.ClientSet)
+		} else if listImages {
+			service.PrintListImages(client.ClientSet)
+		} else {
+			common.ValidateArgs(client.ClientSet, config)
+
+			service.RunE2E(client.ClientSet)
+			client.PrintE2ELogs()
+			client.FetchFiles(config, clientSet, viper.GetString("output-dir"))
+			client.FetchExitCode()
+			service.Cleanup(client.ClientSet)
+		}
+		log.Println("Exiting with code: ", client.ExitCode)
```

Comment:
looks like for printing errors you are using combination of fmt.Print and log statements (eg L71) in this file. is this right? I think we can just stick with log lines

Reply:
Oh I used `fmt` for just testing, thanks for pointing out. Fixed.